plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/DropExtendedStatsProcedure.java

```diff
@@ -82,14 +80,11 @@ public void dropStats(ConnectorSession session, ConnectorAccessControl accessCon
 
         SchemaTableName name = new SchemaTableName(schema, table);
         DeltaLakeMetadata metadata = metadataFactory.create(session.getIdentity());
-        ConnectorTableHandle tableHandle = metadata.getTableHandle(session, name);
+        LocatedTableHandle tableHandle = metadata.getTableHandle(session, name);
         if (tableHandle == null) {
             throw new TrinoException(INVALID_PROCEDURE_ARGUMENT, format("Table '%s' does not exist", name));
         }
-        if (tableHandle instanceof CorruptedDeltaLakeTableHandle corruptedTableHandle) {
```

Comment:
If the table is corrupted, why do we continue the flow now?
This is a functionality change.

Reply:
You're right. 

If you were the procedure anew today (having the `LocatedTableHandle` interface), what code would you write?

Reply:
I'd probably refrain from making any change on the corrupted table.
Changing the state of the extended stats for the table doesn't really improve it.